### PR TITLE
Add support to LUD-17 scheme

### PR DESCRIPTION
**Pull request by ademar111190**
LUD-17: link https://github.com/fiatjaf/lnurl-rfc/blob/luds/17.md

The code running on the app.

Related PR: https://github.com/breez/breezmobile/pull/700

# Channel

https://user-images.githubusercontent.com/1225438/150155040-5eb3692d-c3f4-4ffc-aa49-b03527216bbf.mp4

# Payment

https://user-images.githubusercontent.com/1225438/150155054-e2f118ba-dd78-4f2b-91bb-6824f9f14096.mp4

# Withdraw

https://user-images.githubusercontent.com/1225438/150155068-59849352-ddde-43da-a39f-c0e2ffa90d1c.mp4

### CODE REVIEW

File: account/lnurl.go [hunk 1]
```diff
@@ -118,15 +118,21 @@ func (a *Service) HandleLNURL(rawString string) (result *data.LNUrlResponse, err
...
 		if url, err := url.Parse(rawurl); err == nil {
 			host = url.Host
 		}
 		a.lnurlPayMetadata.encoded = params.EncodedMetadata
-		a.lnurlPayMetadata.data = params.Metadata
+		a.lnurlPayMetadata.data = [][]string{}
 		var metadata []*data.LNUrlPayMetadata
-		for _, e := range params.Metadata {
+
+		var adapter = [][]string{
```
Comment (ademar111190) [inline account/lnurl.go:130]: The go ln url  library has changed the Metadata from a `[][]srting` to a `Metadata` https://github.com/fiatjaf/go-lnurl/blob/master/pay.go#L72

As I'm not able to replicate how the proto buffers models are being generated on the project I did this adaptation with the current model, but I believe we should create a proper model, to do it I need a little help 😬

File: account/lnurl.go [hunk 2]
```diff
@@ -421,18 +427,6 @@ func (a *Service) FinishLNURLPay(params *data.LNURLPayResponse1) (*data.LNUrlPay
...
 
-	/* TODO 9. If routes array is not empty: verifies signature for every provided `ChannelUpdate`, may use these routes if fee levels are acceptable.
-
-	   ref. https://github.com/lightningnetwork/lightning-rfc/blob/master/07-routing-gossip.md#the-channel_update-message
-	   ref. github.com\lightningnetwork\lnd\lnwire\channel_update.go
-
-	   ref. github.com/lightningnetwork/lnd/lnwire/channel_update.go
-	   ref. github.com/lightningnetwork/lnd/lnwire/message.go
-	*/
-	if len(payResponse2.Routes) > 0 {
```
Comment (ademar111190) [inline account/lnurl.go:432]: go-lnurl has changed Routes to `interface{}` so IMHO this code does not make sense anymore

File: account/lnurl.go (hunk 1 above)
Comment (roeierez) [reply account/lnurl.go:130]: two options here:
1. send me the messages.proto file that describes the new model and I will generate the code
2. Contat me offline and we'll try to fix it together. I am available

**Review by roeierez — COMMENTED** [review]
Apart from the model change looks good. I am available to talk offline in telegram.

**Review by roeierez — APPROVED** [review]

**Pull request merged**